firebase-app-distribution/src/main/java/com/google/firebase/app/distribution/TesterSignInClient.java

```diff
@@ -133,16 +139,24 @@ private void onActivityDestroyed(Activity activity) {
               ErrorMessages.APP_BACKGROUNDED,
               FirebaseAppDistributionException.Status.UPDATE_NOT_AVAILABLE));
     }
-    signInTaskCompletionSource = new TaskCompletionSource<>();
+
+    synchronized (signInTaskLock) {
```

Comment:
There is a possible race condition of two threads calling this simultaneously. In that case both threads might bypass the "isCurrentlySigningIn" check above and the second thread might overwrite the first thread and probably induce a weird UX condition. Its probably rare but since we removed method synchronization, it's possible. 

I think a fix for this is to make sure that isSigningIn and re-assignment of the mutable variable to be in the same synchronization block.

Reply:
If you look at `cachedUpdateIfNewReleaseTask` above, we do something similar where the check and the assignment happens in the same synchronization block.

Reply:
Great catch! Updated